airbyte-webapp/src/views/Connection/ConnectionForm/ConnectionForm.tsx

```diff
@@ -118,10 +118,13 @@ const DirtyChangeTracker: React.FC<DirtyChangeTrackerProps> = ({ dirty, onChange
   return null;
 };
 
-interface ConnectionFormProps {
+export type ConnectionOrPartialConnection =
+  | WebBackendConnectionRead
+  | (Partial<WebBackendConnectionRead> & Pick<WebBackendConnectionRead, "syncCatalog" | "source" | "destination">);
```

Comment:
Can you help me understand this type?  I'm trying to understand how we'd end up with a partial connection.  Or is this in anticipation of the PATCH endpoints?

Reply:
https://github.com/airbytehq/airbyte/blob/89ef25a28eb9bf452117bc6ac94d709cb4036451/airbyte-webapp/src/components/CreateConnectionContent/CreateConnectionContent.tsx#L41
When we create a connection we only pass in these properties.